django/applications/catmaid/migrations/0006_add_history_table_support.py

```diff
@@ -0,0 +1,893 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import connection, migrations, models
+from django.conf import settings
+
+
+# Indicate whether history tables are enabled for this migration. If they are
+# not, only history tables will be created, but no triggers are installed and no
+# initial population of history tables will happen.
+history_tracking_enabled = getattr(settings, 'HISTORY_TRACKING', True)
+
+add_history_functions_sql = """
+
+    -- Create a table to keep track of created history tables, when they were
+    -- created, whether triggers were installed on the live table, what the name
+    -- of the live table's primary key is and if the live table has a particular
+    -- column representing time. The latter is used to synchronize tables if
+    -- history tracking is enabled after it was disabled. This table is also used
+    -- for rolling back this migration and more robust access to individual
+    -- history tables based on a live table name.
+    CREATE TABLE catmaid_history_table (
+        history_table_name      name PRIMARY KEY,
+        live_table_name         regclass,
+        triggers_installed      boolean NOT NULL,
+        live_table_time_column  text,
+        live_table_pkey_column  text,
+        creation_time           timestamptz NOT NULL DEFAULT current_timestamp
+    );
+
+
+    -- This enum type represents different types of history changes that are
+    -- recorded in CATMAID's transaction information table.
+    CREATE TYPE history_change_type AS ENUM ('Backend', 'Migration', 'External');
+
+
+    -- The combination of transaction ID and execution time is unique and
+    -- represents one semantic front-end action, initiated by a particular
+    -- user. The unique constraint isn't added explicitelt to avoid performance
```

Comment:
I'm a bit confused by this comment. The `transaction_id` and `execution_time` tuple is unique because of the PK constraint. So we do pay the index maintenance overhead. Considering that `transaction_id` is also indexed in all the history tables, though, this overhead is relatively minor. Also a typo nit.